spec/dpl/providers/pages_spec.rb

```diff
@@ -88,4 +88,15 @@
   describe 'given --email other' do
     it { should have_run 'git config user.email "other"' }
   end
+
+  describe 'given --deploy_key a2V5Cg==', record: true do
+    let(:args) { |e| args_from_description(e) }
+    it { should have_run '[info] Setting up deploy key in ~/.dpl/deploy_key' }
+    it { should have_run '[info] Setting up git-ssh' }
+    it { should have_run '[info] $ ssh -i ~/.dpl/deploy_key -T git@github.com:travis-ci/dpl.git' }
+    it { should have_run 'ssh -i ~/.dpl/deploy_key -T git@github.com:travis-ci/dpl.git' }
+    it { should have_run '[info] Pushing to github.com/travis-ci/dpl.git' }
+    it { should have_run 'git push --quiet "git@github.com:travis-ci/dpl.git" "gh-pages":"gh-pages" > /dev/null 2>&1' }
+    it { should have_run_in_order }
```

Comment:
Could you please also add a test that the SSH-key file is missing after the script has terminated?
Cases to cover:
1) successful completion
2) failure
3) SIGTERM
4) SIGINT